pkg/kubectl/cmd/attach_test.go

```diff
@@ -192,6 +193,72 @@ func TestAttach(t *testing.T) {
 	}
 }
 
+func TestAttachWarnings(t *testing.T) {
+	version := testapi.Default.Version()
+	tests := []struct {
+		name, container, version, podPath, expectedErr, expectedOut string
+		pod                                                         *api.Pod
+		stdin, tty                                                  bool
+	}{
+		{
+			name:        "fallback tty if not supported",
+			version:     version,
+			podPath:     "/api/" + version + "/namespaces/test/pods/foo",
+			pod:         attachPod(),
+			stdin:       true,
+			tty:         true,
+			expectedErr: "Unable to use a TTY - container bar doesn't allocate one",
+		},
+	}
+	for _, test := range tests {
+		f, tf, codec := NewAPIFactory()
+		tf.Client = &fake.RESTClient{
+			Codec: codec,
+			Client: fake.HTTPClientFunc(func(req *http.Request) (*http.Response, error) {
+				switch p, m := req.URL.Path, req.Method; {
+				case p == test.podPath && m == "GET":
```

Comment:
Good question. This is basicallly the same strategy used in `TestAttach` a few lines before.